tools/mo/openvino/tools/mo/main.py

```diff
@@ -310,24 +319,45 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
+    fallback_reasons['extensions'] = \
+        lambda argv : hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+    fallback_reasons['transformations_config'] = \
+        lambda argv: hasattr(argv, 'transformations_config') and argv.transformations_config is not None and len(argv.transformations_config) > 0
+
+    reasons = [reason for reason, is_applicable in fallback_reasons.items() if is_applicable(argv)]
+    return reasons
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
 
+def prepare_ir(argv : argparse.Namespace):
+    argv = arguments_post_parsing(argv)
     t = tm.Telemetry()
     graph = None
     ngraph_function = None
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
-
-    if moc_front_end:
+    fallback_reasons = check_fallback(argv)
+    if moc_front_end and len(fallback_reasons) == 0:
         t.send_event("mo", "conversion_method", moc_front_end.get_name() + "_frontend")
         moc_front_end.add_extension(TelemetryExtension("mo", t.send_event, t.send_error, t.send_stack_trace))
-        ngraph_function = moc_pipeline(argv, moc_front_end)
-    else:
-        t.send_event("mo", "conversion_method", "mo_legacy")
-        graph = unified_pipeline(argv)
+        try:
+            ngraph_function = moc_pipeline(argv, moc_front_end)
+            return graph, ngraph_function
+        except Exception as e:
+            fallback_reasons.append(f"frontend failure with exception: {e}")
```

Comment:
It hides all errors from the new front-end. It shouldn't be. We discussed some predefined list of well described reasons for the fallback. They are listed above in the code in check_fallback. We shouldn't fallback in case of any error in the new front-end.

Reply:
@slyalin  I see your point. I've started the mail thread about it. In order to not block this PR I'm processing it here without handling the exception case.